src/__tests__/TuxMutableRenderMixin-test.js

```diff
@@ -0,0 +1,104 @@
+'use strict';
+
+var moduleToTest = '../TuxMutableRenderMixin';
+
+jest.dontMock(moduleToTest);
+
+describe('TuxMutableRenderMixin', function () {
+  var tuxMutableRenderMixin, nextProps, nextState, tuxMutableTraits;
+
+  beforeEach(function () {
+    tuxMutableRenderMixin = require(moduleToTest);
+    tuxMutableRenderMixin.constructor = {};
+    tuxMutableRenderMixin.props = {
+      message: {
+        text: 'test text'
+      },
+      value: 2
+    };
+    tuxMutableRenderMixin.state = {
+      editing: false
+    };
+    tuxMutableRenderMixin.mutableTraits = {
+      props: ['text', 'value'],
+      state: 'editing'
+    };
+  });
+
+  it('should have componentWillMount and shouldComponentUpdate properties', function () {
+    expect(tuxMutableRenderMixin.componentWillMount).toBeDefined();
+    expect(tuxMutableRenderMixin.shouldComponentUpdate).toBeDefined();
+  });
+
+  describe('componentWillMount', function () {
+    it('should add the __tuxMutableTraits__ prop to the constructor prop', function () {
+      expect(tuxMutableRenderMixin.constructor.__tuxMutableTraits__).toBeUndefined();
+
+      tuxMutableRenderMixin.componentWillMount();
+
+      expect(tuxMutableRenderMixin.constructor.__tuxMutableTraits__).toBeDefined();
+    });
+
+    it('should add the paths to the mutableTraits specified by strings and arrays', function () {
+      tuxMutableRenderMixin.componentWillMount();
+      tuxMutableTraits = tuxMutableRenderMixin.constructor.__tuxMutableTraits__;
+
+      expect(tuxMutableTraits.length).toEqual(3);
+      expect(tuxMutableTraits[0]).toEqual(['props', 'message', 'text']);
+      expect(tuxMutableTraits[1]).toEqual(['props', 'value']);
+      expect(tuxMutableTraits[2]).toEqual(['state', 'editing']);
+    });
+
+    it('should not recreate mutable traits paths if the __tuxMutableTraits__ prop exists ', function () {
+      // Establish initial __tuxMutableTraits__ value
+      tuxMutableRenderMixin.componentWillMount();
+      tuxMutableTraits = tuxMutableRenderMixin.constructor.__tuxMutableTraits__;
+
+      // Create mockComponent and assign the value created during the first componentWillMount call
+      var mockComponent = {};
+      mockComponent.constructor = {};
+      mockComponent.constructor.__tuxMutableTraits__ = tuxMutableTraits;
+      tuxMutableRenderMixin.constructor = mockComponent.constructor;
+
+      // Call componentWillMount a second time, expecting the value of __tuxMutableTraits__ to be equal to the value from the first invocation
+      tuxMutableRenderMixin.componentWillMount();
```

Comment:
This test might be a little bit more semantic if, instead of assigning `tuxMutableRenderMixin` the `constructor` property.  You used `.call` with `componentWillMount` and passed in the `mockComponent` as the context.  That is probably what is happening in `React` anyway.  Just a suggestion.